gmqtt/aioclient.py

```diff
@@ -0,0 +1,91 @@
+import asyncio
+from .client import Client, Message
+from typing import Optional, Tuple
+
+
+class MqttClientProtocol:
+    """
+    Wraps the normal client in the async/await API
+    """
+
+    def __init__(
+        self, inner_client: Client, loop: Optional[asyncio.AbstractEventLoop] = None
+    ):
+        if loop is None:
+            loop = asyncio.get_event_loop()
+        self.loop = loop
+
+        self.cbclient = inner_client
+        self.message_queue = asyncio.Queue()
+
+    async def publish(self, topic: str, message: Message, qos=0):
+        """Publish a message to the MQTT broker"""
+        self.cbclient.publish(topic, message, qos)
+
+    async def subscribe(self, topic: str, qos) -> Tuple[str, bytes]:
```

Comment:
This method is not correct; 

Here example why:
```
# init connection
...
client.subscribe('TEST/#')
message = await client.subscribe('TEST2/#')
```

User will expect `message` from topic "TEST2/#" but this is false: User will get message from one of them.
To fix that behaviour, I suggest you to wait while we finish that functionality (separate callbacks for topics) in core client.

Reply:
Yeah, I'm not happy with the subscription part of the API design yet. A MQTT subscription is long-lived, yet here it looks like receiving a single message. I think a subscription object might be better suited:
```py
sub = await client.subscribe('TEST/#')
sub2 = await client.subscribe('TEST2/#')
message = await sub2.receive()
```
Subscription objects could also have an `.unsubscribe()` method, which was impossible in the original design. I'm sure there will be other functions we want to add to subscriptions later on as well.

I think I can think of a hack to get this working, but the separate callbacks for topics will make this much easier and cleaner code.

Reply:
Alright, I've implemented the hack. I'll discuss it in a comment below.